docs/utilities/validation.md

```diff
@@ -0,0 +1,717 @@
+---
+title: Validation (JSON Schema)
+descrition: Utility
+---
+
+<!-- markdownlint-disable MD043 --->
+
+This utility provides [JSON Schema](https://json-schema.org) validation for events and responses, including JMESPath support to unwrap events before validation.
+
+!!! warning
+    This feature is currently under development. As such it's considered not stable and we might make significant breaking changes before going before its release. You are welcome to [provide feedback](https://github.com/aws-powertools/powertools-lambda-typescript/discussions/3519) and [contribute to its implementation](https://github.com/aws-powertools/powertools-lambda-typescript/milestone/18).
+
+## Key features
+
+- Validate incoming event and response payloads
+- JMESPath support to unwrap events before validation
+- Built-in envelope to unwrap popular AWS service events
+- TypeScript support with type-safe validation
+
+## Getting started
+
+```bash
+npm install @aws-lambda-powertools/validation
+```
+
+You can validate inbound and outbound payloads using the validator class method decorator or Middy.js middleware.
+
+You can also use the standalone `validate` function, if you want more control over the validation process such as handling a validation error.
+
+!!! tip "Using JSON Schemas for the first time?"
+    Check this step-by-step guide on [how to create JSON Schemas](https://json-schema.org/learn/getting-started-step-by-step.html). By default, we support JSON Schema draft-07.
+
+### Validator decorator
+
+The `@validator` decorator is a class method decorator that you can use to validate both the incoming event and the response payload.
+
+If the validation fails, we will throw a `SchemaValidationError`.
+
+??? note "A note on class method decorators"
+    The class method decorators in this project follow the experimental implementation enabled via the [`experimentalDecorators` compiler option](https://www.typescriptlang.org/tsconfig#experimentalDecorators) in TypeScript. We will add support for the newer Stage 3 decorators proposal in the next major release.
+
+    All our decorators assume that the method they are decorating is an async method. This means that even when decorating a synchronous method, it will return a promise. If this is not the desired behavior, you can use one of the other patterns to validate your payloads.
+
+=== "getting_started_decorator.ts"
```

Comment:
We might drop a `TODO` to extract all examples to snippets folder once we have the utility.

Reply:
Yes, we'll also have go through all of them to make sure they match the final implementation. That's why I left them in for now